lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb

```diff
@@ -7,341 +7,230 @@ module Providers
-                genocolorectal.add_gene_colorectal('APC')
-                genocolorectal.add_status(1)
-                genotypes << genocolorectal
-              end
-            elsif colorectal_input.blank? && record.raw_fields['genotype'].blank? &&
-                  TEST_SCOPE_FROM_TYPE_MAP_COLO[moltesttype&.downcase.strip] == :targeted_mutation ||
-                  TEST_SCOPE_FROM_TYPE_MAP_COLO[moltesttype&.downcase.strip].nil?
-              genocolorectal.add_status(:unknown)
-              genotypes << genocolorectal
-            else
-              @logger.debug "FAILED gene parse for  #{colorectal_input}"
-              genocolorectal.add_status(1)
-              genotypes << genocolorectal
+          def positive_rec?(genocolorectal)
+            genocolorectal.attribute_map['teststatus'] == 2
+          end
+
+          def process_targeted_screen(genocolorectal, record, genocolorectals)
+            variant = record.raw_fields['genotype']
+            gene = get_gene(record)
+            genocolorectal.add_gene_colorectal(gene)
+            process_variants(genocolorectal, variant) if positive_rec?(genocolorectal)
+            genocolorectals.append(genocolorectal)
+            genocolorectals
+          end
+
+          def get_gene(record)
+            positive_genes = []
+            gene = record.raw_fields['gene']
+            positive_genes = gene.scan(COLORECTAL_GENES_REGEX).flatten.uniq unless gene.nil?
+            if positive_genes.size.zero?
+              positive_genes = record.raw_fields['investigation code'].
+                               scan(COLORECTAL_GENES_REGEX).flatten.uniq
             end
-            genotypes
+            positive_genes[0] unless positive_genes.nil?
+          end
+
+          def full_screen?(genocolorectal)
+            return false if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Full screen/i).size.positive?
+          end
+
+          def targeted?(genocolorectal)
+            return false if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Targeted/i).size.positive?
           end
 
-          def summarize
-            @logger.info ' ************** Handler Summary **************** '
-            @logger.info "Num bad variants: #{@failed_variant_counter} of "\
-                         "#{@variants_processed_counter} processed"
+          def no_scope?(genocolorectal)
+            return false if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Unable/i).size.positive?
           end
+
+          def pathogenic?(record)
+            varpathclass = record.raw_fields['variantpathclass']&.downcase
+            return true if NON_PATHEGENIC_CODES.exclude? varpathclass
+
+            false
```

Comment:
HI @shilpigoeldev, no need to explicitly return `true` or `false` here, just have `NON_PATHEGENIC_CODES.exclude? varpathclass` be the return value of the method

Reply:
done ✅